Source/Core/Core/PowerPC/BreakPoints.cpp

```diff
@@ -171,6 +173,9 @@ void MemChecks::Add(const TMemCheck& memory_check)
   bool had_any = HasAny();
   if (GetMemCheck(memory_check.start_address) == nullptr)
     m_mem_checks.push_back(memory_check);
+  bool lock = Core::PauseAndLock(true);
+  PowerPC::DBATUpdated();
+  Core::PauseAndLock(false, lock);
```

Comment:
As you pause the emulation, just also clear the jitcache in here instead of using the threadsafe way.